classic/selection/FeatureModel.js

```diff
@@ -14,319 +14,22 @@
-                feat.setStyle();
-            }
-        }
-    },
-
-    /**
-     * Handles the 'singleclick' event of the #map.
-     * Detects if a feature of the connected #layer has been clicked and selects
-     * this feature by selecting its corresponding grid row.
-     *
-     * @private
-     * @param  {ol.MapBrowserEvent} evt OL event object
-     */
-    onFeatureClick: function(evt) {
-        var me = this;
-        var feat = me.map.forEachFeatureAtPixel(evt.pixel,
-            function(feature) {
-                return feature;
-            }, {
-                layerFilter: function(layer) {
-                    return layer === me.layer;
-                }
-            });
-
-        if (feat) {
-            // select clicked feature in grid
-            me.selectMapFeature(feat);
-        }
-    },
-
-    /**
-     * Selects / deselects a feature by triggering the corresponding actions in
-     * the grid (e.g. selecting / deselecting a grid row).
-     *
-     * @private
-     * @param  {ol.Feature} feature The feature to select
-     */
-    selectMapFeature: function(feature) {
-        var me = this;
-        var row = me.store.findBy(function(record, id) {
-            return record.getFeature() == feature;
-        });
-
-        // deselect all if only one can be selected at a time
-        if (me.getSelectionMode() === 'SINGLE') {
-            me.deselectAll();
-        }
-
-        if (feature.get(me.selectedFeatureAttr)) {
-            // deselect feature by deselecting grid row
-            me.deselect(row);
-        } else {
-            // select the feature by selecting grid row
-            if (row != -1 && !me.isSelected(row)) {
-                me.select(row, !this.singleSelect);
-                // focus the row in the grid to ensure it is visible
-                me.view.focusRow(row);
-            }
-        }
-    },
-
-    /**
-     * Overwrites the onSelectChange function of the father class.
-     * Ensures that the selected feature is added / removed to / from
-     * #selectedFeatures lookup object.
-     *
-     * @private
-     * @param  {GeoExt.data.model.Feature} record Selected / deselected record
-     * @param  {Boolean} isSelected Record is selected or deselected
-     */
-    onSelectChange: function(record, isSelected) {
-        var me = this;
-        var selFeature = record.getFeature();
-
-        // toggle feature's selection state
-        selFeature.set(me.selectedFeatureAttr, isSelected);
-
-        if (isSelected) {
-            me.selectedFeatures.push(selFeature);
-        } else {
-            me.selectedFeatures.remove(selFeature);
-        }
-
-        me.callParent(arguments);
-    },
-
-    /**
-     * Overwrites parent's destroy method in order to unregister the OL events,
-     * that were added on init.
-     * Needed due to the lack of destroy event of the parent class.
-     *
-     * @private
-     */
-    destroy: function() {
```

Comment:
As I see, all functions of this class moved to the mixin. However, `destroy` did not. Is this intentionally?

Reply:
the only thing that the destroy function did was calling `unbindOlEvents` so I replaced it with:
```
mixinConfig: {
  after: {
    destroy: 'unbindOlEvents'
...
```

Reply:
but maybe we still need the `destroying` flag ...

Reply:
i readded that